example/src/VideoroomContext.tsx

```diff
@@ -0,0 +1,41 @@
+import * as React from 'react';
+
+type Action = { type: 'set'; val: string };
+type Dispatch = (action: Action) => void;
+type State = { roomName: string };
+
+const VideoroomContext = React.createContext<
+  { state: State; dispatch: Dispatch } | undefined
+>(undefined);
+
+function VideoroomReader(state, action) {
+  switch (action.type) {
+    case 'set':
+      return { roomName: action.val };
+    default:
+      throw new Error(`Unhandled action type: ${action.type}`);
+  }
+}
+
+const VideoroomContextProvider = (props) => {
+  const [state, dispatch] = React.useReducer(VideoroomReader, {
```

Comment:
Ok, now why did you use `useReducer` here? `useState` was ok and simpler here :D In the example there was a counter that you could decrement and increment so I guess it was more complicated but here you just set and read room name. `useReducer` is used for complicated state.

So the context would like like this:
```
const [roomName, setRoomName] = useState()
...
<Provider value={{roomName, setRoomName}}>
...
```
And then to use it you simply do something like this:
```
const { roomName, setRoomName } = useRoomName()
```
Much simpler and obvious imho. no weird `state` and `dispatch`.

Also we can use this context for more than just a room name (for example I guess user name would be useful). So maybe instead `useRoomName` hook, we can have something like `useVideoroomState` or something.

Reply:
Okey, I changed it to be useState now, I guess this provider isn't essential right now and app could work without it, but prob. it will be useful soon so I guess it can stay.